Automattic/jetpack
# Fix URL parsing issues / cache contamination

When a requested URL has a `//` after the hostname, GET parameters may be ignored, causing cache contamination across URLs.

This PR fixes the places where partial URLs are parsed using `parse_url`, ensuring they are read properly.

Fixes p3btAN-1Sl-p2

#### Changes proposed in this Pull Request:
* Use `$_SERVER['QUERY_STRING']` when checking if GET params exist, and
* Introduce a new wrapper for `parse_url` which deals with partial URLs that may have double-slashes in them.

#### Other information:

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Turn on wp-super-cache, and clear your cache.
* Run `curl http://your-domain.com//?s=1223`
* Visit `http://your-domain.com/` as a non-logged in user, and observe that the search results page is served.
* Turn on the "Tracking Parameters" feature in the Advanced tab.
* Run `curl 'http://your-domain.com//?s=1&mtm_cid=123'`. Observe the cache time in the footer.
* Run `curl 'http://your-domain.com//?s=1&mtm_cid=12'`. Observe the cache time in the footer, and ensure it is unchanged (i.e.: cached version changed, as the tracking param is ignored).
